File: PizzaTime.Data/Pizzeria.cs
namespace PizzaTime.Data;

public class Pizzeria
{
    public string Email { get; set; } = "";
    public string Piva { get; set; } = "";
    public string Address { get; set; } = "";

    public static Pizzeria Generate(string iva)
    {
        Pizzeria pizzeria = new Pizzeria();
        pizzeria.Email = "testmail@test";
        pizzeria.Piva = iva;
        pizzeria.Address = "testAddress, p.address";
        return pizzeria;
    }
}

File: PizzaTimeSPA/src/data/Pizzeria.ts
import { Config, names, starWars, uniqueNamesGenerator } from "unique-names-generator";

export default class Pizzeria{
    public email : string = "";
    public piva: string = "";
    public address: string = "";

    public static generate(piva: string) : Pizzeria{
        const config : Config = {dictionaries:[names]};
        const st : Config = {dictionaries:[starWars]};
        var email = `${uniqueNamesGenerator(config)}.pizzeria@${uniqueNamesGenerator(st)}`;
        var address = `via ${uniqueNamesGenerator(config)} ${Math.floor(Math.random()*100)}`;
        return {email,piva,address};
    }
};